Does not compile against node.js 22.12.0

I'm using your node-snap7 module in my [homebridge ](https://homebridge.io/) plugin [homebridge-plc](https://github.com/Feilner/homebridge-plc). In the **2024-12-12** version of the latest [docker containers for homebridge](https://hub.docker.com/r/homebridge/homebridge/) the node version was updated to 22.12.0.

When installing and thus compiling against node.js 22.12.0 snap7 and node-snap7 are generating some some warnings.
But unfortunately node-snap7 fails 


```
In file included from ../src/node_snap7_client.h:11,
                 from ../src/node_snap7.cpp:6:
../../nan/nan.h: In function 'void Nan::SetAccessor(v8::Local<v8::ObjectTemplate>, v8::Local<v8::String>, GetterCallback, SetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute, imp::Sig)':
../../nan/nan.h:2548:19: error: no matching function for call to 'v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>&, void (*&)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value>&), void (*&)(v8::Local<v8::Name>, v8::Local<v8::Value>, const v8::PropertyCallbackInfo<void>&), v8::Local<v8::Object>&, v8::AccessControl&, v8::PropertyAttribute&)'
 2548 |   tpl->SetAccessor(
      |   ~~~~~~~~~~~~~~~~^
 2549 |       name
      |       ~~~~         
 2550 |     , getter_
      |     ~~~~~~~~~      
 2551 |     , setter_
      |     ~~~~~~~~~      
 2552 |     , obj
      |     ~~~~~          
 2553 |     , settings
      |     ~~~~~~~~~~     
 2554 |     , attribute
      |     ~~~~~~~~~~~    
 2555 | #if (NODE_MODULE_VERSION < NODE_16_0_MODULE_VERSION)
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 2556 |     , signature
      |     ~~~~~~~~~~~    
 2557 | #endif
      | ~~~~~~             
 2558 |   );
      |   ~

```


```

In file included from /home/homebridge/.cache/node-gyp/22.12.0/include/node/v8-function.h:15,
                 from /home/homebridge/.cache/node-gyp/22.12.0/include/node/v8.h:33,
                 from /home/homebridge/.cache/node-gyp/22.12.0/include/node/node.h:73,
                 from ../src/node_snap7_client.h:10:
/home/homebridge/.cache/node-gyp/22.12.0/include/node/v8-template.h:1049:8: note: candidate: 'void v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>, v8::AccessorGetterCallback, v8::AccessorSetterCallback, v8::Local<v8::Value>, v8::PropertyAttribute, v8::SideEffectType, v8::SideEffectType)'
 1049 |   void SetAccessor(
      |        ^~~~~~~~~~~
/home/homebridge/.cache/node-gyp/22.12.0/include/node/v8-template.h:1052:61: note:   no known conversion for argument 5 from 'v8::AccessControl' to 'v8::PropertyAttribute'
 1052 |       Local<Value> data = Local<Value>(), PropertyAttribute attribute = None,
      |                                           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/homebridge/.cache/node-gyp/22.12.0/include/node/v8-template.h:1055:8: note: candidate: 'void v8::ObjectTemplate::SetAccessor(v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::Local<v8::Value>, v8::PropertyAttribute, v8::SideEffectType, v8::SideEffectType)'
 1055 |   void SetAccessor(
      |        ^~~~~~~~~~~
/home/homebridge/.cache/node-gyp/22.12.0/include/node/v8-template.h:1058:61: note:   no known conversion for argument 5 from 'v8::AccessControl' to 'v8::PropertyAttribute'
 1058 |       Local<Value> data = Local<Value>(), PropertyAttribute attribute = None,
      |                                           ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
../../nan/nan.h: In function 'void Nan::SetAccessor(v8::Local<v8::ObjectTemplate>, v8::Local<v8::String>, GetterCallback, SetterCallback, v8::Local<v8::Value>, v8::AccessControl, v8::PropertyAttribute)':
../../nan/nan.h:2594:19: error: no matching function for call to 'v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>&, void (*&)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value>&), void (*&)(v8::Local<v8::Name>, v8::Local<v8::Value>, const v8::PropertyCallbackInfo<void>&), v8::Local<v8::Object>&, v8::AccessControl&, v8::PropertyAttribute&)'
 2594 |   tpl->SetAccessor(
      |   ~~~~~~~~~~~~~~~~^
 2595 |       name
      |       ~~~~         
 2596 |     , getter_
      |     ~~~~~~~~~      
 2597 |     , setter_
      |     ~~~~~~~~~      
 2598 |     , obj
      |     ~~~~~          
 2599 |     , settings
      |     ~~~~~~~~~~     
 2600 |     , attribute
      |     ~~~~~~~~~~~    
 2601 |   );
      |   ~                
/home/homebridge/.cache/node-gyp/22.12.0/include/node/v8-template.h:1049:8: note: candidate: 'void v8::ObjectTemplate::SetAccessor(v8::Local<v8::String>, v8::AccessorGetterCallback, v8::AccessorSetterCallback, v8::Local<v8::Value>, v8::PropertyAttribute, v8::SideEffectType, v8::SideEffectType)'
 1049 |   void SetAccessor(
      |        ^~~~~~~~~~~

```

Full compile log can be seen here:
[homebridge-plc-error.log](https://github.com/user-attachments/files/18254978/homebridge-plc-error.log)

In the past there was a similar issue with node 19 #88 
Which was fixed by the two PRs: #93 and #94

--- a/.github/workflows/test-and-release.yml
+++ b/.github/workflows/test-and-release.yml
@@ -8,7 +8,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        node: [16, 18, 20]
+        node: [16, 18, 20, 22]
         arch: [x86, x64]
         exclude:
           - { os: ubuntu-latest, arch: x86 }

--- a/README.md
+++ b/README.md
@@ -5,7 +5,7 @@
 [![Test and Release](https://github.com/mathiask88/node-snap7/actions/workflows/test-and-release.yml/badge.svg)](https://github.com/mathiask88/node-snap7/actions/workflows/test-and-release.yml)
 [![npm](https://img.shields.io/npm/dm/node-snap7.svg?label=dl)](https://www.npmjs.com/package/node-snap7)
 
-**Current node-snap7 version:** 1.0.7\
+**Current node-snap7 version:** 1.0.8\
 **Current snap7 version:** 1.4.2
 
 **In my spare time I am working on a [node-addon-api](https://github.com/nodejs/node-addon-api) rewrite and want to switch from [prebuild-install](https://github.com/prebuild/prebuild-install) to [prebuildify](https://github.com/prebuild/prebuildify).\

--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "node-snap7",
   "main": "./lib/node-snap7.js",
-  "version": "1.0.7",
+  "version": "1.0.8",
   "description": "Native node.js addon/wrapper for snap7",
   "homepage": "https://github.com/mathiask88/node-snap7",
   "repository": {
@@ -18,17 +18,17 @@
     "name": "Mathias Küsel"
   },
   "engines": {
-    "node": ">=16"
+    "node": "^16 || ^18 || ^20 || ^22"
   },
   "license": "MIT",
   "readmeFilename": "README.md",
   "dependencies": {
-    "nan": "~2.18.0",
-    "bindings": "~1.5.0",
-    "prebuild-install": "^7.1.1"
+    "nan": "^2.22.0",
+    "bindings": "^1.5.0",
+    "prebuild-install": "^7.1.2"
   },
   "devDependencies": {
-    "prebuild": "^12.1.0",
+    "prebuild": "^13.0.1",
     "prebuild-ci": "^4.0.0"
   },
   "scripts": {

